src/features/landing/components/AuthSection/index.tsx

```diff
@@ -23,9 +30,45 @@ function AuthSection() {
     formState: { errors },
   } = useForm<RegisterItem>();
   const onSubmit = (data: RegisterItem) => {
-
     console.log(data);
   };
+
+  const isLengthValid = (value: string) =>
+    value.length >= 8 && value.length <= 64;
+
+  const hasNumber = (value: string) => /[0-9]/.test(value);
+
+  const hasUppercase = (value: string) => /[A-Z]/.test(value);
+
+  function validatePassword(value: string) {
+    if (!isLengthValid(value))
+      return 'Password must be between 8 and 64 characters long';
+    if (!hasNumber(value)) return 'Password must contain at least one number';
+    if (!hasUppercase(value))
+      return 'Password must contain at least one uppercase letter';
+    return true;
+  }
+
+  function isValidEmailFormat(email: string) {
+    return /^.+@.+\..+$/.test(email);
+  }
+
+  function validateEmail(email: string) {
+    if (!isValidEmailFormat(email)) return 'Invalid email format';
+    return true;
```

Comment:
string or boolean? this is complicated, is the validate option from react-hook-form require a string or boolean?